### PR TITLE
basic tests for API calls to backend complete

### PR DESCRIPTION
**Pull request by hongmm**
Created two tests to check both API endpoints. Data is hard coded so that database is not queried. Both tests pass with the current implementation, tests might need to be refactored later on since I am still not clear on exactly how to write asyc tests properly.

### CODE REVIEW

**Review by Ahmed-Elawad — COMMENTED** [review]
@hongmm These additions LGTM. Some of your code in _mocks_/mockIndex(response method)  & _tests_/server-spec(module imports) looks commented out. Is that intentional?

**Comment by hongmm** [comment]
@Ahmed-Elawad some of the code is commented out because I was playing around with jest looking for solutions, I have a bad habit of leaving stuff commented out and not deleting them. the response section I will fill out at some other point, right now the comment is there as a filler.

File: __mocks__/mockIndex.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+const reviews = [
+    {
+      "id":1,
```
Comment (jonathandavidlewis) [inline __mocks__/mockIndex.js:3]: The double quotes are unnecessary. looks like json. If you wanted, you could indeed make a json file review.json and instead import that. If it's a js file, it's distracting to have the double-quotes.

File: __mocks__/mockIndex.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      } else {
+        const productId = Number(url.substr(1));
+        reviews[0].product_id === productId ? resolve(reviews) : reject({error: `product id ${productId} does not exist`});
+      }
+      reject({error: 'something went wrong in the test'});
+    })
+  })
+};
+
+var response = function() {
```
Comment (jonathandavidlewis) [inline __mocks__/mockIndex.js:36]: All the comments are a no no. It's distracting for someone trying to understand the implications of merging in your code. If you have code you are thinking about adding later, you can save it off in a git branch or a git stash locally.

**Pull request merged**